src/Commands/DrushCommands.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Drupal\migrate_generator\Commands;
+
+use Drush\Commands\DrushCommands as DrushCommandsBase;
+use Drupal\migrate_generator\Scanner;
+use Drupal\migrate_generator\Generator;
+
+/**
+ * Defines Drush commands for the migrate_generator module.
+ */
+class DrushCommands extends DrushCommandsBase {
+
+  /**
+   * Migrate source scanner.
+   *
+   * @var \Drupal\migrate_generator\Scanner
+   */
+  protected $scanner;
+
+  /**
+   * Migrate source generator.
+   *
+   * @var \Drupal\migrate_generator\Generator
+   */
+  protected $generator;
+
+  /**
+   * Constructor method.
+   *
+   * @param \Drupal\migrate_generator\Scanner $scanner
+   *   Class for migrate source scanner.
+   * @param \Drupal\migrate_generator\Generator $generator
+   *   Class for migrate source generator.
+   */
+  public function __construct(Scanner $scanner, Generator $generator) {
+    parent::__construct();
+    $this->scanner = $scanner;
+    $this->generator = $generator;
+  }
+
+  /**
+   * Generate migrations based on source CSVs.
+   *
+   * @param string $directory
+   *   Path to directory with source .csv files.
+   * @param array $options
+   *   Additional options for the command.
+   *
+   * @option pattern
+   *   Pattern for source CSV files. Defaults to '*.csv'.
+   * @option delimiter
+   *   Delimiter for source CSV files. Defaults to ;
+   * @option enclosure
+   *   Enclosure for source CSV files. Defaults to "
+   * @option values_delimiter
+   *   Delimiter for multi-valued fields. Defaults to |
+   * @option date_format
+   *   Date format used in CSV. Defaults to "d-m-Y H:i:s"
+   * @option update
+   *   Update previously-generated migrations.
+   *
+   * @command migrate_generator:generate_migrations
+   *
+   * @usage drush migrate_generator:generate_migrations $(pwd)/web/modules/custom/migrate_generator/example
+   *   Generate migrations from CSV sources from example folder.
+   */
+  public function generateMigrations($directory, array $options = [
+    'pattern' => '*.csv',
+    'delimiter' => ';',
+    'enclosure' => '"',
+    'values_delimiter' => '|',
+    'date_format' => 'd-m-Y H:i:s',
+  ]) {
+    // Scan source files.
+    $sources = $this->scanner->readSources($directory, $options);
+    if ($sources) {
+      $migration_group = $this->generator->createMigrationGroup();
+      foreach ($sources as $entity_type => $entity_info) {
+        foreach ($entity_info as $bundle => $source_info) {
+          $migration = $this->generator->createMigration($entity_type, $bundle, $source_info, $migration_group, $options);
+          if ($migration) {
+            $this->logger()->notice('Migration ' . $migration->id() . ' was created');
+          }
+        }
+      }
+    }
+    $this->logger()->notice(
+      'All migrations were successfully generated. You can now run them using next command:
```

Comment:
Would it be possible to contextualize this message ? It can be a bit un-true like in this case where not all migrations were generated, only some: 
![image](https://user-images.githubusercontent.com/20738264/83277249-0eb26d80-a1d2-11ea-9608-6f32958c5e49.png)

We could have _All migrations were successfully generated_ when none of the files failed to generate it's migration and another message like `Some migrations were successfully generated` when not all of the files failed to generate it's migration but some did

Reply:
I have refactored messages a bit.